metricflow/dataflow/builder/dataflow_plan_builder.py

```diff
@@ -450,8 +503,35 @@ def _find_measure_recipe(
                 for y in x.join_on_partition_time_dimensions
             )
 
+            measure_node = node_with_lowest_cost
+
+            # Retrieves the grouped non_additive_dimension_spec and agg_time_dimension
+            measure_properties = self._build_measure_spec_properties(measure_specs)
```

Comment:
This is nice! I would love this to be part of the method signature, maybe containing the measure spec sequence as well. So this method would take in a MeasureSpecPropeties or MeasureSpecsByCommonProperties or whatever we end up calling it, and it's obvious to all of us that we want a collected set of measures. Maybe we could even put in the data source name and collect these at the very top, and then remove all of the validation checks inside of this method.

This is getting into separate PR territory, though, so I think it's fine as-is. Maybe add a TODO comment if you don't pull this into the method signature straightaway.

Reply:
Did this in this PR!